### DISCUSSION TITLE
Novice Build - Will my GPIO wiring work?

### DISCUSSION THREAD

**Discussion by drezelrs** [MiczFlor/RPi-Jukebox-RFID#1732] [General]
I am planning my build and as a novice I want to make sure my wiring is correct. 
See the images below for hardware and my diagrams so far. Will my wiring of the RFID reader and GPIO buttons work?

Hardware:

- The GPIO expansion board is so I don't have to solder the OnOff Shim directly onto the Pi. This apparently avoids undervoltage issues as seen here: [https://splittscheid.de/phoniebox-bauanleitung-toniebox-alternative/#ftoc-das-dilemma-mit-dem-ausschalten](url)
- OnOff Shim is so that I can implement a safe shutdown on/off rocker switch.
- RasPiO analog zero is to wire a slide potentiometer for volume control.

![Slide6](https://user-images.githubusercontent.com/93059901/149633041-13064385-ad1c-4135-8518-3c9f46d5e412.PNG)

This is how I believe I will connect all the boards:
![Slide7](https://user-images.githubusercontent.com/93059901/149633058-827d02ed-8768-49d6-8087-85f9b63f8b80.PNG)

Wiring RC522 RFID Reader:
I want to use an RFID reader that allows for stop on removal. I plan on leaving the RFID card on the reader and removing to stop play.
![Slide8](https://user-images.githubusercontent.com/93059901/149633060-d9c35220-99c1-4df6-a346-9e14e96f22c6.PNG)

Wiring RC522 RFID Reader and some GPIO buttons:
![Slide9](https://user-images.githubusercontent.com/93059901/149633066-7ab8cfa2-fa28-4249-8a95-40fb27d2d225.PNG)

Next step, assuming all above is correct (which I am sure it likely is not), is to wire the rocker switch to the OnOff Shim and the slide potentiometer to the RasPiO analog zero.

**Comment by lroeser** [comment]
Hello,
you can find the wiring for the RC522 card-reader here: https://github.com/MiczFlor/RPi-Jukebox-RFID/wiki/Wiring_for_RC522_card_reader
(consider the alternative wiring with a single 4x2 pin block).

Information on button wiring you can find here:
https://github.com/MiczFlor/RPi-Jukebox-RFID/wiki/Using-GPIO-hardware-buttons

OnOff shim you should find multiple HowTos online as this is not phoniebox specific.

I guess for the RasPi Analog Zero and the volume slider you are on your own...

**Reply by drezelrs** [reply]
Thanks for the info. I am new to GPIO so I want to really know how I can look for interferences. The RFID is wired exactly how the documentation requires, but I want to know if the OnOffShim is going to block or interfere with that. This is why I think I need the expansion board, but again I am new to this so not sure if it will work. 

I've ordered all the parts and will be prototyping soon anyways so I guess I'll know hands on soon enough.

**Reply by lenlennart** [reply]
RFID readers use very different pins than OnOff shim, what makes you think they interfer?

Edit: Ahh, i think i know where the confusion is coming from. You can use any power and ground pin for either component. As started above, consider the alternative wiring for the rc522.
